uefi/src/proto/shell/mod.rs

```diff
@@ -2,12 +2,64 @@
 
 //! EFI Shell Protocol v2.2
 
-use crate::proto::unsafe_protocol;
+use uefi_macros::unsafe_protocol;
 
-pub use uefi_raw::protocol::shell::ShellProtocol;
+use core::ptr;
+
+use uefi_raw::protocol::shell::ShellProtocol;
+
+use crate::{CStr16, Char16, Result, StatusExt};
 
 /// Shell Protocol
 #[derive(Debug)]
 #[repr(transparent)]
-#[unsafe_protocol(uefi_raw::protocol::shell::ShellProtocol::GUID)]
-pub struct Shell(uefi_raw::protocol::shell::ShellProtocol);
+#[unsafe_protocol(ShellProtocol::GUID)]
+pub struct Shell(ShellProtocol);
+impl Shell {
+    /// Returns the current directory on the specified device
+    ///
+    /// # Arguments
+    ///
+    /// * `file_system_mapping` - The file system mapping for which to get
+    ///   the current directory
+    ///
+    /// # Returns
+    ///
+    /// * `Some(cwd)` - CStr16 containing the current working directory
+    /// * `None` - Could not retrieve current directory
+    #[must_use]
+    pub fn current_dir(&self, file_system_mapping: Option<&CStr16>) -> Option<&CStr16> {
+        let mapping_ptr: *const Char16 = file_system_mapping.map_or(ptr::null(), CStr16::as_ptr);
+        let cur_dir = unsafe { (self.0.get_cur_dir)(mapping_ptr.cast()) };
+        if cur_dir.is_null() {
+            None
+        } else {
+            unsafe { Some(CStr16::from_ptr(cur_dir.cast())) }
+        }
+    }
+
+    /// Changes the current directory on the specified device
+    ///
+    /// # Arguments
+    ///
+    /// * `file_system` - Pointer to the file system's mapped name.
+    /// * `directory` - Points to the directory on the device specified by
+    ///   `file_system`.
+    ///
+    /// # Returns
+    ///
+    /// * `Status::SUCCESS` - The directory was successfully set
```

Comment:
Drop the `Returns` section. It's understood that `Ok` is returned in the successful case.